src/app/recover-funds/[address]/RecoverFundsButton.tsx

```diff
@@ -0,0 +1,183 @@
+'use client';
+import React, { useState } from 'react';
+import { useMemo } from 'react';
+import {
+  Address,
+  getL2Network,
+  L1ToL2MessageGasEstimator,
+  L1TransactionReceipt,
+} from '@arbitrum/sdk';
+import { Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory';
+import { getBaseFee } from '@arbitrum/sdk/dist/lib/utils/lib';
+import { useNetwork, useSigner } from 'wagmi';
+import { getProviderFromChainId, getTargetChainId } from '@/utils';
+import { BigNumber } from 'ethers';
+
+function RecoverFundsButton({
+  balanceToRecover,
+  chainID,
+  destinationAddress,
+  aliasedAddress,
+}: {
+  balanceToRecover: BigNumber;
+  chainID: number;
+  destinationAddress: string;
+  aliasedAddress: string;
+}) {
+  const [message, setMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+  const { chain } = useNetwork();
+  const { data: signer } = useSigner({ chainId: chain?.id });
+
+  const redeemButton = useMemo(() => {
+    if (!signer) return null;
+
+    if (chain?.id !== 1 && chain?.id !== 5) {
+      return (
+        <div>Unknown L1 chain id. This chain is not supported by this tool</div>
+      );
+    }
+
+    if (chain?.id !== chainID) {
+      return (
+        <div>
+          To recover funds, connect to chain ${chain?.id} (${chain?.name})
+        </div>
+      );
+    }
+
+    return (
+      <div className="recover-funds-form">
+        <button
+          id="recover-button"
+          disabled={loading}
+          onClick={async () => {
+            setLoading(true);
+            setMessage('');
+
+            // Start transaction
+            const signerAddress = new Address(await signer.getAddress());
+
+            // We instantiate the Inbox factory object to make use of its methods
+            const targetChainID = getTargetChainId(chainID);
+
+            const baseL2Provider = getProviderFromChainId(targetChainID);
+            const l2Network = await getL2Network(baseL2Provider);
```

Comment:
`useNetwork`?

Reply:
This would not work, we want to instantiate l2Network regardless of current network